public/app/features/explore/utils/links.ts

```diff
@@ -60,3 +61,23 @@ function getTitleFromHref(href: string): string {
   }
   return title;
 }
+
+function buildHostUrl() {
+  return `${window.location.protocol}//${window.location.host}${config.appSubUrl}`;
+}
+
+function getRelativeURLPath(url: string) {
+  let path = url.replace(buildHostUrl(), '');
+  return path.startsWith('/') ? path.substring(1, path.length) : path;
+}
+
+export const createShortLink = memoizeOne(async function(path: string) {
+  try {
+    const shortUrl = await getBackendSrv().post(`/api/short-urls`, {
+      path: getRelativeURLPath(path),
+    });
+    return shortUrl.url;
+  } catch (err) {
```

Comment:
I would not catch it here. As @Elfo404  mentioned I would rethrow and then show alert.error for the user, considering this is server call there are more things that could go wrong.